runner/sidecar/source/db/db.go

```diff
@@ -0,0 +1,220 @@
+package db
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	dfv1 "github.com/argoproj-labs/argo-dataflow/api/v1alpha1"
+	"github.com/argoproj-labs/argo-dataflow/runner/sidecar/source"
+	sharedutil "github.com/argoproj-labs/argo-dataflow/shared/util"
+	_ "github.com/go-sql-driver/mysql"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+var logger = sharedutil.NewLogger()
+
+const offsetTableSchema = `CREATE TABLE IF NOT EXISTS argo_dataflow_offsets (
+	table_name VARCHAR(255) NOT NULL,
+	consumer_group VARCHAR(255) NOT NULL,
+	offset VARCHAR(255),
+	PRIMARY KEY(table_name, consumer_group)
+)`
+
+type rowData = map[string]interface{}
+
+type dbSource struct {
+	db *sql.DB
+}
+
+func New(ctx context.Context, secretInterface corev1.SecretInterface, clusterName, namespace, pipelineName, stepName string, replica int, sourceName string, x dfv1.DBSource, f source.Func) (source.Interface, error) {
```

Comment:
In the future - we need a way, like S3, to load balance across the replicas. Probably only the lead replica (replica=0) should query and make HTTP callback to service with data. This allows us to, in future, process in parallel. 

I think we need shared code for this. I need to think more on the topic.